### PR TITLE
pnpm-lock-export: init at 0.4.0

### PR DESCRIPTION
**Pull request by ambroisie**
###### Description of changes

A program that can generate `yaml.lock` files from `pnpm-lock.yaml`.

I intend on using it when [`Woodpecker`](https://woodpecker-ci.org/) updates to its next version, as they have migrated to using `pnpm`.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/web/pnpm-lock-export/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  passthru = {
+    updateScript = ./update.sh;
+  };
+
+  meta = with lib; {
+    description = "A utility for converting pnpm-lock.yaml to other lockfile formats";
+    homepage = "https://github.com/cvent/pnpm-lock-export";
+    license = licenses.mit;
+    platforms = platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/development/web/pnpm-lock-export/default.nix:29]:
```suggestion
```
this defaults to nodejs.meta.platforms

File: pkgs/development/web/pnpm-lock-export/update.sh [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    TOKEN_ARGS=(--header "Authorization: token $GITHUB_TOKEN")
+fi
+
+if [[ $# -gt 1 || $1 == -* ]]; then
+    echo "Regenerates packaging data for the pnpm-lock-export packages."
+    echo "Usage: $0 [git release tag]"
+    exit 1
+fi
+
+set -x
```
Comment (SuperSandro2000) [inline pkgs/development/web/pnpm-lock-export/update.sh:16]:
```suggestion
set -eou pipefail
```

File: pkgs/development/web/pnpm-lock-export/default.nix (hunk 1 above)
Comment (ambroisie) [reply pkgs/development/web/pnpm-lock-export/default.nix:29]: TIL. Done.

File: pkgs/development/web/pnpm-lock-export/update.sh (hunk 2 above)
Comment (ambroisie) [reply pkgs/development/web/pnpm-lock-export/update.sh:16]: It's already there, a few lines down, to avoid the error on `version=$1`.

**Pull request merged**